include/trampolines.h

```diff
@@ -85,9 +93,17 @@ dmtcp_setup_trampoline_by_addr(void *addr,
 
   /* Base address of page where func resides. */
   page_base = (void *)((unsigned long)info->addr & pagemask);
-
+  int pagecount = 1;
+
+  /* Increase the pagecount if number of bytes needs to be wriiten
+   * to set a trampoline falls in the next page.
+   */
+  if (((unsigned long)page_base + pagesize) - (unsigned long)info->addr
+        < sizeof(data.bytes)) {
```

Comment:
Shouldn't this code be:
 `if (((unsigned long)page_base + pagesize) - (unsigned long)info->jump < ASM_JUMP_LEN) {`

(We want to replace ASM_JUMP_LEN bytes and not simply `sizeof(data.bytes)` number of bytes.)

Reply:
Yes, you are right. `ASM_JUMP_LEN` is the right length for this edge case. I will be changing the `if` condition in the next commit.